common/include/cstdint

```diff
@@ -0,0 +1,37 @@
+#ifndef __CSTDINT_H__
+#define __CSTDINT_H__
+
+#include <stdint.h>
+
+namespace std {
+using int8_t = ::int8_t;
+using int16_t = ::int16_t;
+using int32_t = ::int32_t;
+using int64_t = ::int64_t;
+using int_fast8_t = ::int_fast8_t;
+using int_fast16_t = ::int_fast16_t;
+using int_fast32_t = ::int_fast32_t;
+using int_fast64_t = ::int_fast64_t;
+using int_least8_t = ::int_least8_t;
+using int_least16_t = ::int_least16_t;
+using int_least32_t = ::int_least32_t;
+using int_least64_t = ::int_least64_t;
+using intmax_t = ::intmax_t;
+using intptr_t = ::intptr_t;
+using uint8_t = ::uint8_t;
+using uint16_t = ::uint16_t;
+using uint32_t = ::uint32_t;
+using uint64_t = ::uint64_t;
+using uint_fast8_t = ::uint_fast8_t;
+using uint_fast16_t = ::uint_fast16_t;
+using uint_fast32_t = ::uint_fast32_t;
+using uint_fast64_t = ::uint_fast64_t;
+using uint_least8_t = ::uint_least8_t;
+using uint_least16_t = ::uint_least16_t;
+using uint_least32_t = ::uint_least32_t;
+using uint_least64_t = ::uint_least64_t;
+using uintmax_t = ::uintmax_t;
+using uintptr_t = ::uintptr_t;
+} // namespace std
+
+#endif
```

Comment:
I think that the it's debatable whether an interrupt constitutes concurrent execution. In this scenario, it looks like other implementations consider this a recursion problem since the two entries into the __cxa_guard_acquire can't be concurrent.

It was interesting to see what clang and gcc do with the code below. Clang/libc++ aborts with `libc++abi.dylib: __cxa_guard_acquire detected recursive initialization`  and gcc (on linux) throws an undocumented internal exception type with a similar message.


```
#include <string_view>
#include <signal.h>
#include <unistd.h>
#include <stdio.h>

using namespace std::literals::string_view_literals;

int count_construction = 0;
volatile bool recvd_sigusr1 = false;

class Singleton
{
public:
	static Singleton & getInstance()
	{
		static Singleton instance;
		return instance;
	}

	bool is_initialized() const noexcept { return m_initialized; }
private:
	Singleton() noexcept
	{
		count_construction += 1;

		while (!recvd_sigusr1) {}

		m_initialized = true;
	}

	bool m_initialized = false;
};

void puts_signal_safe(std::string_view msg)
{
	::write(1, msg.data(), msg.length());
	::write(1, "\n", 1);
}

void on_usr1(int)
{
	recvd_sigusr1 = true;
	auto & instance = Singleton::getInstance();
	if (count_construction > 1)
	{
		puts_signal_safe("constructed more than once!"sv);
	}
	if (!instance.is_initialized())
	{
		puts_signal_safe("not initialized!"sv);
	}
}

int main(int argc, char * argv[])
{
	signal(SIGUSR1, on_usr1);

	printf("Send SIGUSR1 to %d\n", ::getpid());

	auto & instance = Singleton::getInstance();
	puts_signal_safe("returned from getInstance"sv);
	if (count_construction > 1)
	{
		puts_signal_safe("constructed more than once!"sv);
	}
	if (!instance.is_initialized())
	{
		puts_signal_safe("instance not initialized!"sv);
	}
	return 0;
}
```